transcom/mymove
# INT B-19831 use Okta DODID and disable customer DODID edit

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A946595)

## Summary

Currently we have a risk of the customer's DODID being different than the one provided in Okta. Since the merge of requiring CAC validation on initial registration for customers, we now have an accurate DODID for the customer stored in Okta so we don't want this to be changed by the service member.

This PR does the following:
- updated the values when creating a new user through `devlocal` to include `cac_edipi` and use email formats for `username` and `login` profile values
- uses the `cac_edipi` value stored in the `oktaUser` state in the DODID text on initial customer registration (this is what saves to `service_members.edipi`
- disables the input of the DODID inputs in the `DoDInfoForm`, `EditOktaInfoForm`, and `ServiceInfoForm` components
- updated all relevant tests
- updated relevant storybooks to reflect changes


### How to test

1. Access MM as a customer
2. When registering for a profile, you'll see that the DODID field is disabled and populated with a value (if authenticating with CAC it should be that DODID - if using devlocal, it should be `1234567890`)
3. Create a move and upload orders
4. Go to edit your profile & edit the `Service Info` section
5. You'll see that you can view, but not update your DODID - form still submits
6. Now go back to edit your profile and edit your Okta profile
7. You'll see that you can view, but not update your Okta DODID - form still submits

## Screenshots
<img width="594" alt="Screenshot 2024-05-03 at 11 07 51 AM" src="https://github.com/transcom/mymove/assets/136510600/e683a719-20ae-4258-a600-e4f3e8de3266">

<img width="838" alt="Screenshot 2024-05-03 at 11 40 25 AM" src="https://github.com/transcom/mymove/assets/136510600/b6eacb8a-1a22-4209-8235-f9f73931b95d">

<img width="677" alt="Screenshot 2024-05-03 at 11 40 37 AM" src="https://github.com/transcom/mymove/assets/136510600/a69e6ec3-f032-4d4e-b381-74222350e111">
